### ISSUE TITLE
[Process] Consider allowing streamable input

### ISSUE THREAD

**Issue by schmittjoh** [symfony/symfony#14482]
Currently, the process component assumes that input is known before a command is invoked. Sometimes, you want to stream the input, i.e. gradually write it to the process that is being run. This could for example be done by exposing the pipes, or maybe better a specific method for writing:

``` php
$process = new Process('takes-some-input');
$process->start();

while ($stream->hasContent()) {
   $process->write($stream->read());
}
```

**Comment by sstok** [comment]
I like this idea :+1: and I already have a good use-case: Git rebase interactive.

**Comment by javiereguiluz** [comment]
Closing it as fixed by #18386.
